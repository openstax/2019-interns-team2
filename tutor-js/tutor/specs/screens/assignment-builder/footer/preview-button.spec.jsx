import Preview from '../../../../src/screens/assignment-builder/footer/preview-button';

describe('Task Plan Builder: Preview button', () => {

  let props;
  beforeEach(() => {
    props = {
      planType: 'reading',
      isWaiting: false,
      isNew: true,
      courseId: '1',
    };
  });

  it('renders when plan hw or reading', () => {
    const btn = shallow(<Preview {...props} />);
    expect(btn.html()).not.toBeNull();
    btn.setProps({ planType: 'external' });
    expect(btn.html()).toBeNull();
    btn.setProps({ planType: 'homework' });
    expect(btn.html()).not.toBeNull();
    btn.unmount();
  });

  it('matches snapshot', function() {
    expect.snapshot(<Preview {...props} />).toMatchSnapshot();
  });

});
