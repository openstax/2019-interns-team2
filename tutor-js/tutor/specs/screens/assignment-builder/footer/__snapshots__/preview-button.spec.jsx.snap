// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Plan Builder: Preview button matches snapshot 1`] = `null`;
